common/page.go

```diff
@@ -364,6 +363,127 @@ func (p *Page) initEvents() {
 	}()
 }
 
+func (p *Page) urlGroupingName(ctx context.Context, urlTag string) (string, bool) {
```

Comment:
As far as I can see, a new reader (e.g., me) doesn't know what this method does or what its purpose is in general. I haven't seen it explained anywhere. I'd add a comment here to explain its purpose and how it works concisely.

Reply:
Sure, added in 34f3eaab5dbec984cff25f1f626924831a699a52

Reply:
Thanks!